var/spack/repos/builtin/packages/umpire/package.py

```diff
@@ -82,7 +82,7 @@ class Umpire(CachedCMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("cmake@3.8:", type="build")
     depends_on("cmake@3.9:", when="+cuda", type="build")
-    depends_on("cmake@:3.20", when="+rocm", type="build")
+    depends_on("cmake@:3.20", when="+examples +rocm", type="build")
```

Comment:
I believe this is an instance of

> The AMD ROCm Platform hipcc compiler was identifed by CMake 3.21.0 through 3.21.2 as a distinct compiler with id ROCMClang. This has been removed because it caused regressions.

So it's probably fine to drop the `depends_on` by now or change it into

```suggestion
    conflicts("^cmake@3.21.0:3.21.2", when="+rocm")
```

Reply:
I think this is different. At least I still get the error in the examples reported by #27873 with CMake 3.23.2.

However, that restriction you're talking about may also be necessary...

Reply:
I also tried building with CMake 3.21.0 and it actually seems to work so for some reason umpire is immune to that problem. Umpire does not (explicitly?) set `hipcc` as the C++ compiler which is probably why it's not affected.